spring-cloud-stream-app-starters-docs/src/main/asciidoc/overview.adoc

```diff
@@ -1,8 +1,185 @@
+This is done by running the application generation Maven plugin.
+You can do so by simply invoking the corresponding script in the root of the project.
+
+```
+./generate.sh
+```
+
+The script will generate for the each of the prebuilt applications the following items:
+
+* `pom.xml` file with the required dependencies (starter and binder)
+* a class that contains the `main` method of the application and imports the predefined configuration
+* generated integration test code that exercises the component against the configured binder.
+
+For example, `spring-cloud-starter-stream-sink-cassandra` will generate `cassandra-sink-rabbit` and `cassandra-sink-kafka` as completely functional applications.
+
+=== Creating custom artifacts
+
+Apart from accessing the sources, sinks and processors already provided by the project.
+In this section we will describe how to:
+
+* Use a different binder than Kafka or Rabbit
+* Create your own applications and starters
+* Customize dependencies such as Hadoop distributions or JDBC drivers
+
+==== Using a different binder
+
+If you want to use one of the applications found in Spring Cloud Stream Application Starters and you want to use one of the predefined binders (i.e. Kafka or Rabbit), you can just use the prebuilt versions of the artifacts.
+But if you want to connect to a different middleware system, and you have a binder for it, you will create new artifacts.
+
+[source,xml]
+----
+<dependencies>
+  <!- other dependencies -->
+  <dependency>
+    <group>org.springframework.cloud.stream.app</group>
+    <artifactId>spring-cloud-starter-stream-sink-cassandra</artifactId>
+    <version>1.0.0.BUILD-SNAPSHOT</version>
+  </dependency>
+  <dependency>
+    <group>org.springframework.cloud</group>
+    <artifactId>spring-cloud-stream-binder-gemfire</artifactId>
+    <version>1.0.0.BUILD-SNAPSHOT</version>
+  </dependency>
+</dependencies>
+----
+
+The next step is to create the project's main class and import the configuration provided by the starter.
+For example, in the same case of the Cassandra sink it can look like the following:
+
+[source,java]
+----
+package org.springframework.cloud.stream.app.cassandra.sink.rabbit;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.stream.app.cassandra.sink.CassandraSinkConfiguration;
+import org.springframework.context.annotation.Import;
+
+
+@SpringBootApplication
+@Import(CassandraSinkConfiguration.class)
+public class CassandraSinkGemfireApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(CassandraSinkGemfireApplication.class, args);
+	}
+}
+----
+
+==== Creating your own applications and starters
+
+Spring Cloud Stream Application Starters consists of regular Spring Cloud Stream applications, with some additional conventions that facilitates generating prebuilt applications with the preconfigured binders.
```

Comment:
“Spring Cloud Stream Application Starters consists of regular Spring Cloud Stream applications with some additional conventions that facilitate generating prebuilt applications with the preconfigured binders.”